### PR TITLE
Add C++ solution to cses 2428

### PR DESCRIPTION
**Pull request by DarkMenacer**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [done] I have tested my code.
- [done] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [done] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
https://usaco.guide/general/adding-solution?lang=cpp

File: solutions/cses-2428.mdx [hunk 1]
```diff
@@ -74,4 +74,60 @@ public class cses2428 {
 
 </JavaSection>
 
+<CPPSection>
+
+```cpp
+/*
+Pranav Chatur
```
Comment (brad182) [inline solutions/cses-2428.mdx:81]: This part probably isn't necessary

File: solutions/cses-2428.mdx [hunk 2]
```diff
@@ -74,4 +74,60 @@ public class cses2428 {
...
+    ll i = 0, j = 1;
+    map<ll,ll> distinct;
+    distinct[integers[i]]++;
+    while(1){
+        if(j==integers.size()+1){
+	/*
+	When j reaches end of the integers, there is no need to subtract 'total subarrays from i+1 to j',
+	(as no new element is going to be added further), therefore answer+= (j-i)*(j-i-1)/2
+	*/
+            answer += (j-i)*(j-i-1)/2;
```
Comment (brad182) [inline solutions/cses-2428.mdx:102]: if you're going to use a space here, make it consistent between things like +-*/

File: solutions/cses-2428.mdx [hunk 3]
```diff
@@ -74,4 +74,60 @@ public class cses2428 {
...
+
+ll give_me_answer(vll &integers, ll k){
+    ll answer = 0;
+    ll i = 0, j = 1;
+    map<ll,ll> distinct;
+    distinct[integers[i]]++;
+    while(1){
+        if(j==integers.size()+1){
+	/*
+	When j reaches end of the integers, there is no need to subtract 'total subarrays from i+1 to j',
```
Comment (brad182) [inline solutions/cses-2428.mdx:99]: use 
/*
 *
*/
formatting

**Review by brad182 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**